bike_rent/models/sale.py

```diff
@@ -11,17 +11,23 @@ def action_confirm(self):
         # <AndriusD code
         def filter_rests(rec):
             product = rec.product_id
-            return product.is_bike and product.type == 'service'
-
-        for record in self.order_line.filtered(filter_rests):  # /AndriusD code>
-            rent_rec = self.env['bike.rent'].create({
-                'notes': record.name,
-                'price': record.price_total,
-                'partner_id': record.order_partner_id.id,
-                'bike_id': record.product_id.id,
-                'rent_time': record.product_id.rent_duration,
-            })
-            record.bike_rent_id = rent_rec.id
+            return product.is_bike and product.type == 'service'  # /AndriusD code>
+
+        def create_bike_rent_objects(so_record):
+            for record_line in so_record.product_id.rented_bike_ids:
+                rent_rec = self.env['bike.rent'].create({
+                    'notes': so_record.product_id.description_sale,
+                    'price': record_line.lst_price,
+                    'partner_id': so_record.order_partner_id.id,
+                    'bike_id': record_line.id,
+                    'rent_time': so_record.product_id.rent_duration,
+                    'sale_id': so_record.order_id.id
+                })
+                record.bike_rent_id = rent_rec.id
+
+        for record in self.order_line.filtered(filter_rests):  # AndriusD code line
+            create_bike_rent_objects(record)
```

Comment:
Multiple sale orders can get passed here, I checked, so maping of lines in needed here. If you were to use my advice, here you would only need to call that method:
```
for line in self.mapped('order_line').filtered(filter_rests):  # AndriusD code line
    line.create_bike_rents()
```